src/rich-text/commands/index.ts

```diff
@@ -82,15 +82,23 @@ export function toggleBlockType(
     nodeType: NodeType,
     attrs?: { [key: string]: unknown }
 ) {
-    const nodeCheck = nodeTypeActive(nodeType, attrs);
-    const setBlockTypeCommand = setBlockType(nodeType, attrs);
-
     return (state: EditorState, dispatch: (tr: Transaction) => void) => {
-        // if the node is set, toggle it off
-        if (nodeCheck(state)) {
+        const nodeCheck = nodeTypeActive(nodeType, attrs);
```

Comment:
Rather than modify the generic `toggleBlockType` function, let's create a `toggleHeadingLevel` (that ultimately will likely call `toggleBlockType`) and use that instead